Sources/Plasma/PubUtilLib/plModifier/plDetectorLog.cpp

```diff
@@ -42,15 +42,21 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 
 #include "plDetectorLog.h"
 
-#ifdef PLASMA_EXTERNAL_RELEASE
 plStatusLog* plDetectorLog::fLog = nullptr;
-#else
-plStatusLog* plDetectorLog::fLog = plStatusLogMgr::GetInstance().CreateStatusLog(
-    20,
-    "Detector.log",
-    plStatusLog::kFilledBackground | plStatusLog::kDeleteForMe |
-    plStatusLog::kDontWriteFile | plStatusLog::kAlignToTop);
+
+void plDetectorLog::EnsureLogInit()
+{
+    if (fLog)
+        return;
+
+#ifndef PLASMA_EXTERNAL_RELEASE
+    fLog = plStatusLogMgr::GetInstance().CreateStatusLog(
+        20,
+        "Detector.log",
+        plStatusLog::kFilledBackground | plStatusLog::kDeleteForMe |
+        plStatusLog::kDontWriteFile | plStatusLog::kAlignToTop);
 #endif
+}
```

Comment:
It looks like this is addressing a side effect of `plProduct` not being initialized in `CreateStatusLog()`. Perhaps `plProduct` should make sure it is initialized when `CreateStatusLog()` calls into it. Otherwise, we still have potential static initialization issues.

Reply:
As far as I can tell, `plProduct` doesn't need to be explicitly initialized in any way - it's all global constants (`static const`). The problem is that (roughly speaking) C++ doesn't specify in what order the global variables of different source files are initialized, so you cannot safely have one global initializer (e. g. `plDetectorLog::fLog`) use another global variable initialized in another file e. g. `plProduct::PRODUCT_LONG_NAME`).

Moving the initialization of `plDetectorLog::fLog` into a method is a sufficient fix, as far as I can tell. As long as the method is never called (even indirectly) from any global initializer, it's guaranteed that all global variables will have been initialized by the time the method can be called.